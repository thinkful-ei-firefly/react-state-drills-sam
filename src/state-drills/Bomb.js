import React from 'react';

class Bomb extends React.Component {

  state = {
    count: 0
  }

  componentDidMount() {
    this.timer = setInterval(() => {
      this.setState({count: this.state.count+1});
      this.state.count>=8 && clearInterval(this.timer);
    }, 1000)
  }

  componentWillUnmount() {
    clearInterval(this.timer)
  }
  

  render() {
    return (
      <div className='bomb'>
        <p>{this.state.count>=8 ? 'Boom!!!!' : (this.state.count%2===0 ? 'tick' : 'tock')}</p>
      </div>
    );
  }
}

export default Bomb;