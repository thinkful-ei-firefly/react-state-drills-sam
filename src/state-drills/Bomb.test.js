import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'
import Bomb from './Bomb';


describe ('Bomb testing', () => {
  it ('renders without crashing', () => {
    const div=document.createElement('div');
    ReactDOM.render(<Bomb />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it ('renders this UI as expected', () => {
    const tree=renderer.create(<Bomb />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});