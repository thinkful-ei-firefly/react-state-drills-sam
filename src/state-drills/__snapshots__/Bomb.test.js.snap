// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bomb testing renders this UI as expected 1`] = `
<div
  className="bomb"
>
  <p>
    tick
  </p>
</div>
`;
